Tests/MathLib/TestGaussLegendreIntegration.cpp

```diff
@@ -0,0 +1,585 @@
+
+    virtual ~LocalAssemblerDataInterface() = default;
+};
+
+template <typename ShapeFunction, typename IntegrationMethod,
+          unsigned GlobalDim>
+class LocalAssemblerData : public LocalAssemblerDataInterface
+{
+    using ShapeMatricesType = ShapeMatrixPolicyType<ShapeFunction, GlobalDim>;
+    using ShapeMatrices = typename ShapeMatricesType::ShapeMatrices;
+
+public:
+    LocalAssemblerData(MeshLib::Element const& e,
+                       std::size_t const /*local_matrix_size*/,
+                       bool is_axially_symmetric,
+                       unsigned const /*integration_order*/)
+        : _e(e)
+    {
+        if (is_axially_symmetric)
+            OGS_FATAL("Only testing Cartesian meshes!");
+    }
+
+    double integrate(const Function& f,
+                     unsigned const integration_order) const override
+    {
+        double integral = 0;
+
+        auto const sms =
+            ProcessLib::initShapeMatrices<ShapeFunction, ShapeMatricesType,
+                                          IntegrationMethod, GlobalDim>(
+                _e, false /*is_axially_symmetric*/,
+                IntegrationMethod{integration_order});
+        IntegrationMethod integration_method{integration_order};
+
+        for (unsigned ip = 0; ip < sms.size(); ++ip)
+        {
+            // We need to assert that detJ is constant in each element in order
+            // to be sure that in every element that we are integrating over the
+            // effective polynomial degree is the one we expect.
+            EXPECT_NEAR(sms[0].detJ, sms[ip].detJ,
+                        std::numeric_limits<double>::epsilon());
+            auto const& N = sms[ip].N;
+            auto const coords = interpolateNodeCoordinates(_e, N);
+            auto const function_value = f(coords);
+            integral += function_value * sms[ip].detJ *
+                        integration_method.getWeightedPoint(ip).getWeight();
+        }
+
+        return integral;
+    }
+
+private:
+    MeshLib::Element const& _e;
+};
+
+class TestProcess
+{
+public:
+    using LocalAssembler = LocalAssemblerDataInterface;
+
+    TestProcess(MeshLib::Mesh const& mesh, unsigned const integration_order)
+        : _integration_order(integration_order),
+          _mesh_subset_all_nodes(mesh, &mesh.getNodes())
+    {
+        std::vector<MeshLib::MeshSubsets> all_mesh_subsets;
+        all_mesh_subsets.emplace_back(&_mesh_subset_all_nodes);
+
+        _dof_table = std::make_unique<NumLib::LocalToGlobalIndexMap>(
+            std::move(all_mesh_subsets), NumLib::ComponentOrder::BY_COMPONENT);
+
+        // createAssemblers(mesh);
+        ProcessLib::createLocalAssemblers<LocalAssemblerData>(
+            mesh.getDimension(), mesh.getElements(), *_dof_table, 1,
+            _local_assemblers, mesh.isAxiallySymmetric(), _integration_order);
+    }
+
+    double integrate(LocalAssembler::Function const& f) const
+    {
+        double integral = 0;
```

Comment:
I don't have a good reference for that, but I think the compiler will implicitly convert the 0 to a double 0.0
So I'd keep that code. If you insist, I can change it.